x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -336,9 +336,93 @@
                             .indices(".enrich-*")
                             .privileges("manage", "read", "write")
                             .build() }, null, MetadataUtils.DEFAULT_RESERVED_METADATA))
+                .put("viewer", buildViewerRoleDescriptor())
+                .put("editor", buildEditorRoleDescriptor())
                 .immutableMap();
     }
 
+    private static RoleDescriptor buildViewerRoleDescriptor() {
+        return new RoleDescriptor("viewer",
+            new String[] {},
+            new RoleDescriptor.IndicesPrivileges[] {
+                // Stack
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("/~(([.]|ilm-history-).*)/")
+                    .privileges("read", "view_index_metadata").build(),
+                // Security
+                RoleDescriptor.IndicesPrivileges.builder().indices(".siem-signals-*").privileges("read", "view_index_metadata").build() },
+            new RoleDescriptor.ApplicationResourcePrivileges[] {
+                RoleDescriptor.ApplicationResourcePrivileges.builder().application("kibana-.kibana").resources("*").privileges(
+                    "feature_discover.read",
+                    "feature_dashboard.read",
+                    "feature_canvas.read",
+                    "feature_maps.read",
+                    "feature_ml.read",
+                    "feature_graph.read",
+                    "feature_visualize.read",
+                    "feature_logs.read",
+                    "feature_infrastructure.read",
+                    "feature_apm.read",
+                    "feature_uptime.read",
+                    "feature_siem.read",
+                    "feature_dev_tools.read",
+                    "feature_advancedSettings.read",
+                    "feature_indexPatterns.read",
+                    "feature_savedObjectsManagement.read",
+                    "feature_savedObjectsTagging.read",
+                    "feature_fleet.read",
+                    "feature_actions.read",
+                    "feature_stackAlerts.read").build() },
+            null,
+            null,
+            MetadataUtils.DEFAULT_RESERVED_METADATA,
+            null);
+    }
+
+    private static RoleDescriptor buildEditorRoleDescriptor() {
+        return new RoleDescriptor("editor",
```

Comment:
@bytebilly would we expect the `editor` role to be able to create/update/delete Kibana spaces? Or is this a task reserved for administrators?

If we need the editor to manage spaces, then this role would instead need to grant the `all` application privilege, and the `reserved_ml_admin` privilege (`all` does not automatically grant ML access until 8.0):

```
RoleDescriptor.ApplicationResourcePrivileges.builder()
                    .application("kibana-.kibana")
                    .resources("*").privileges("all")
                    .build()
```

```
RoleDescriptor.ApplicationResourcePrivileges.builder()
                    .application("kibana-*")
                    .resources("*").privileges("reserved_ml_admin")
                    .build()
```

Reply:
@legrego I'm not sure to follow your comment. We are not using the `all` shortcut but rather assigning all individual permissions (happy to simplify if we can do differently).

Could you also explain why there is `kibana-.kibana` in the first row, and `kibana-*` in the second one? Thanks!

Reply:
Sorry for not being clear -- If we want editors to be able to manage spaces, then we have to use the `all` shortcut in order to achieve this. We don't have a dedicated "manage spaces" privilege today: https://github.com/elastic/kibana/issues/51759, and so it's currently gated behind this `all` privilege

Reply:
> Could you also explain why there is kibana-.kibana in the first row, and kibana-* in the second one? Thanks!

That is an implementation detail of Kibana's "reserved privileges", and is something we can potentially simplify in 8.x once Kibana drops support for multi-tenant setups. The `-*` at the end indicates that this privilege will be granted to all Kibana tenants, not just the default `.kibana` tenant.

Reply:
Larry and I had a sync on that, and he is checking if using his proposed approach will give UX problems or not. Based on that, we can define if we want to use one approach or the other.

Reply:
I didn't get a chance to verify this today. I struggled to get ES to build locally, and my time was quite limited today. I'm on PTO until Wednesday (after feature freeze), so please don't hold this PR up on me. We can tweak the application privileges definition following FF if we have to without any real interruption